docs/configuration.md

```diff
@@ -114,12 +114,67 @@ in the `spark-defaults.conf` file. A few configuration keys have been renamed si
 versions of Spark; in such cases, the older key names are still accepted, but take lower
 precedence than any instance of the newer key.
 
-Spark properties mainly can be divided into two kinds: one is related to deploy, like
-"spark.driver.memory", "spark.executor.instances", this kind of properties may not be affected when
-setting programmatically through `SparkConf` in runtime, or the behavior is depending on which
-cluster manager and deploy mode you choose, so it would be suggested to set through configuration
-file or `spark-submit` command line options; another is mainly related to Spark runtime control,
-like "spark.task.maxFailures", this kind of properties can be set in either way.
+Note that Spark properties have different effective timing and they can be divided into three kinds:
+<table class="table">
+<tr><th>Configuration Type</th><th>Meaning</th><th>Examples</th></tr>
+<tr>
+  <td><code>Configurations needed at driver launch</code></td>
```

Comment:
I would use titles, for example as below instead of "Configurations needed at driver launch" or "Application Deploy Related Configuration"

- Configuration: Launching Driver
- Configuration: Application Deployment

We could also think about having a parent group "Configuration" and place them under it.

Reply:
Update to 
```
Configuration: Launching Driver
Configuration: Application Deployment
```